### PR TITLE
Feat/registration backend

### PR DESCRIPTION
**Pull request by jeromes123**
**OVERVIEW**
Created the backend for registration

**SNIPPETS**
image

### CODE REVIEW

**Comment by sunkenquest** [comment]
Kindly resolve first the conflict.

**Review by sunkenquest — APPROVED** [review]
GEFN 🐑

**Comment by jeromes123** [comment]
**OVERVIEW**
Added custom email verification

**SNIPPETS**

https://github.com/XPERIA679/microblog-ojt1/assets/123230015/c109e62b-a118-4aec-acf6-3bdd5af6c2e1

File: src/app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+use App\Models\User;
+use App\Models\Profile;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use App\Http\Requests\RegisterRequest;
+
+class UserController extends Controller
+{
+    public function register(RegisterRequest $request)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: 1.) Add docbloc without param and returns
2.) Add return type of the method

File: src/app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            'user_id' => $user->id,
+        ]);
+        
+        auth()->login($user);
+        $user->sendEmailVerificationNotification();
+
+        return view('components.auth.verify-email');
+    }
+    
+    public function logout()
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: 1.) Add docbloc without param and returns
2.) Add return type of the method

File: src/app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        auth()->logout();
+        return redirect('/');
+    }
+
+    public function verifyEmail($id, $hash)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: 1.) Add docbloc without param and returns
2.) Add param types and return type of the method

File: src/app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            if (!$user->hasVerifiedEmail()) {
+                $user->markEmailAsVerified();
+            }
+            return view('test-login');
+        } else {
+            Log::error('User not found');
+        }
+    }
+
+    public function sendVerificationNotification(Request $request)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: 1.) Add docbloc without param and returns
2.) Add return type of the method

File: src/app/Models/Profile.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    protected $fillable = [
+        'user_id',
+        'first_name',
+        'last_name',
+        'birth_date',
+        'address',
+        'contact'
+    ];
+
+    public function user(): HasOne
```
Comment (hipe01mark) [inline src/app/Models/Profile.php:25]: 1.) Add docbloc without param and returns

File: src/app/Providers/AuthServiceProvider.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+use Illuminate\Auth\Notifications\VerifyEmail;
+use Illuminate\Notifications\Messages\MailMessage;
+use Illuminate\Support\ServiceProvider;
+
+class AuthServiceProvider extends ServiceProvider
+{
+    /**
+     * Register any authentication / authorization services.
+     *
+     * @return void
```
Comment (hipe01mark) [inline src/app/Providers/AuthServiceProvider.php]: remove!

File: src/app/Http/Controllers/UserController.php (hunk 1 above)
Comment (jeromes123) [reply src/app/Http/Controllers/UserController.php]: Done

File: src/app/Http/Controllers/UserController.php (hunk 2 above)
Comment (jeromes123) [reply src/app/Http/Controllers/UserController.php]: Done

File: src/app/Http/Controllers/UserController.php (hunk 4 above)
Comment (jeromes123) [reply src/app/Http/Controllers/UserController.php]: Done

File: src/app/Models/Profile.php (hunk 5 above)
Comment (jeromes123) [reply src/app/Models/Profile.php:25]: Done

File: src/app/Providers/AuthServiceProvider.php (hunk 6 above)
Comment (jeromes123) [reply src/app/Providers/AuthServiceProvider.php]: Done

**Comment by sunkenquest** [comment]
Kindly resolve the conflict.

**Review by sunkenquest — APPROVED** [review]
LGTM 🙇🏼

**Pull request merged**